pkg/mcp/client/client.go

```diff
@@ -338,6 +342,7 @@ func (c *Client) Run(ctx context.Context) {
 						c.reporter.RecordRecvError(err, status.Code(err))
 						scope.Errorf("Error receiving MCP response: %v", err)
 					}
+					retry = time.After(resendMCPRequestDelay)
```

Comment:
I think that is the desired effect? 

1) Move the scope of `retry` outside of both loops
2) The initial connection should be immediate (nanosecond)
3) Subsequent attempts, either due to connection failure or send/receive failure should be delayed to avoid flooding the server.

wdyt?